prime_numbers.py

```diff
@@ -0,0 +1,31 @@
+def generate_primes(n=100):
+    """
+    This function generates all prime numbers between 1 to n.
+    A prime number is a natural number greater than 1 that has no positive divisors other than 1 and itself.
+    The numbers 0 and 1 are not prime numbers.
+
+    :param n: The upper limit till where to generate prime numbers. Default is 100.
+    :return: A list of prime numbers.
+    """
+    primes = []
+    for possiblePrime in range(2, n + 1):
+        isPrime = True
+        for num in range(2, int(possiblePrime ** 0.5) + 1):
+            if possiblePrime % num == 0:
+                isPrime = False
+                break
```

Comment:
The inner loop runs from 2 to the square root of the number. This can be optimized by checking divisibility only up to the smallest prime number greater than the square root. This can be achieved by storing the prime numbers and checking divisibility only with those primes.